diff.py

```diff
@@ -15,18 +15,39 @@ def diff_line(i, j):
 def compare_source_files_objects(obj1, obj2):
     diff_funcs = []
     if obj1['name'] == obj2['name']:
-        if obj1['functions'] != obj2['functions'] or obj1['coverage'] != obj2['coverage']:
-            if obj1['functions'] == obj2['functions']:
-                obj1['functions'] = []
-            if obj1['functions'] != obj2['functions']:
-                for func1 in obj1['functions']:
-                    for func2 in obj2['functions']:
-                        if func1['name'] == func2['name']:
-                            if func1['exec'] is False and func2['exec'] is True:
-                                diff_funcs.append(func2)
-                obj1['functions'] = diff_funcs
+        are_funcs_equal = False
+        is_cov_equal = False
+
+        if obj1['coverage'] == obj2['coverage']:
+            is_cov_equal = True
+        if obj1['functions'] == obj2['functions']:
+            are_funcs_equal = True
```

Comment:
Instead of doing this, you can store func diff in diff_funcs and line diff in diff_lines and then at the end of the function:
```
if len(diff_funcs) == 0 and len(diff_lines) == 0:
    return None
else:
    obj2['coverage'] = diff_lines
    obj2['functions'] = diff_funcs
    return obj2
```